webpack/assets/javascripts/react_app/redux/actions/common/forms.js

```diff
@@ -89,5 +89,6 @@ export const submitForm = ({
         errorToast,
       })
     );
+    return uniqueAPIKey;
```

Comment:
I don't know if I like it more than I dislike it? lol
in this way we are continuing to use the callbacks which we just dropped in favor of the Redux store,
how about passing the `APIKey` to both `submitForm` and `ForemanForm` at the top?

Reply:
I don't like it either hehe, I think passing the key-item twice makes it better